includes/classes/Feature/Search/Synonyms.php

```diff
@@ -385,13 +385,13 @@ public function add_search_synonyms( $mapping, $index ) {
 	 * @return void
 	 */
 	public function handle_update_synonyms() {
-		$nonce   = filter_input( INPUT_POST, $this->get_nonce_field(), FILTER_SANITIZE_STRING );
-		$referer = filter_input( INPUT_POST, '_wp_http_referer', FILTER_SANITIZE_STRING );
+		$nonce   = filter_input( INPUT_POST, $this->get_nonce_field(), FILTER_SANITIZE_SPECIAL_CHARS );
+		$referer = filter_input( INPUT_POST, '_wp_http_referer', FILTER_SANITIZE_URL );
```

Comment:
1. A nonce should not contain special chars anyway; if it does, it will fail to verify
2. Referrer is meant to be an URL; that's why the `FILTER_SANITIZE_URL` filter seems appropriate here.